doc/doxygen/src/creating-boards.md

```diff
@@ -0,0 +1,293 @@
+
+```c
+/**
+@defgroup    boards_foo FooBoard
+@ingroup     boards
+@brief       Support for the foo board
+
+### User Interface
+
+  ....
+
+### Using UART
+
+  ...
+
+### Flashing the device
+
+  ...
+
+*/
+```
+
+## Using Common code                                        {#common-board-code}
+
+To avoid code duplication, common code across boards has been grouped in
+`boards/common`. e.g. `BOARD`s based on the same cpu (`boards/common/nrf52`) or
+`BOARD`s having the same layout `boards/common/nucleo64`.
+
+In the case of source files this means some functions like `board_init` can be
+already defined in the common code. Unless having specific configurations or
+initialization you might not need a`board.c` or `board.h`. Another common use
+case is common peripheral configurations:
+
+```diff
+-#include "cfg_timer_tim5.h"
++/**
++ * @name   Timer configuration
++ * @{
++ */
++static const timer_conf_t timer_config[] = {
++    {
++        .dev      = TIM5,
++        .max      = 0xffffffff,
++        .rcc_mask = RCC_APB1ENR_TIM5EN,
++        .bus      = APB1,
++        .irqn     = TIM5_IRQn
++    }
++};
++
++#define TIMER_0_ISR         isr_tim5
++
++#define TIMER_NUMOF         ARRAY_SIZE(timer_config)
++/** @} */
+
+```
+
+If you want to use common makefiles, include them at the end of the specific
+`Makefile`, e.g. for a `Makefile.features`:
+
+```mk
+CPU = foo
+CPU_MODEL = foobar
+
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_i2c
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_uart
+
+include $(RIOTBOARD)/common/foo_common/Makefile.features
+```
+
+## Boards outside of RIOTBASE                      {#boards-outside-of-riotbase}
+
+All `BOARD`s in RIOT reside in `RIOTBOARD` (`RIOTBOARD` being a make variable
+set to `$(RIOTBOARD)/boards`).
+
+If one want's to use a `BOARD` outside of `RIOTBOARD`, the way to go is setting
+the `BOARDSDIR` variable to the path to the directory containing your external
+boards, e.g.: `BOARDSDIR=/home/external-boards/` (this would commonly be done
+in your application `Makefile` or your environment).
+
+```
+/home/
+|----RIOT/
+    |---- ...
+|----external-boards/
+    |----board-foo/
+        |----dist/
+            |----scripts
+        |----board.c
+        |----doc.txt
+        |----include/
+            |----periph_conf.h
+            |----board.h
+            |----gpio_params.h
+        |----Makefile
+        |----Makefile.dep
+        |----Makefile.features
+        |----Makefile.include
+```
+
+When defining boards outside of `RIOTBASE` some considerations must be taken
+with the makefiles:
+
+
+- `Makefile`
+  - `MODULE` cannot be `board`: if inheriting from a board `foo-parent`
+    then `foo-parent` already defines a `MODULE = board`, so
+    use any other name, lets say `board-foo`.
+  - Include the location of the parent `BOARD` to inherit from (if there is
+    one):
```

Comment:
I find it a bit confusing that at first this talks about boards outside of `RIOTBASE`, but then gives instructions on what do do if a board inherits from another one.
One does not necessitate the other, does it?

Reply:
Yes, I think you are right that those two cases might not be well differentiated, you can have:

- boards outside RIOTBASE
- boards outside RIOTBASE inheriting code from boards in RIOTBOARD, I'll try to re-phrase